functests/dpdk/dpdk.go

```diff
@@ -79,6 +81,93 @@ var _ = Describe("dpdk", func() {
 			Expect(cpuNumaNode).To(Equal(pciNumaNode))
 		})
 	})
+
+	Context("Validate HugePages", func() {
+		var activeNumberOfFreeHugePages int
+		var numaNode int
+		var dpdkPod *corev1.Pod
+
+		BeforeEach(func() {
+			podList, err := client.Client.Pods(testDpdkNamespace).List(metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, pod := range podList.Items {
+				if pod.OwnerReferences == nil || len(pod.OwnerReferences) == 0 {
+					err = client.Client.Pods(testDpdkNamespace).Delete(pod.Name, &metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+
+			dpdkPod = findDPDKWorkloadPod()
+			buff, err := pods.ExecCommand(client.Client, *dpdkPod, []string{"cat", "/sys/fs/cgroup/cpuset/cpuset.cpus"})
+			Expect(err).ToNot(HaveOccurred())
+			cpuList := strings.Split(strings.Replace(buff.String(), "\r\n", "", 1), ",")
+			numaNode, err = findNUMAForCPUs(dpdkPod, cpuList)
+			Expect(err).ToNot(HaveOccurred())
+
+			buff, err = pods.ExecCommand(client.Client, *dpdkPod, []string{"cat",
+				fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-1048576kB/free_hugepages", numaNode)})
+			Expect(err).ToNot(HaveOccurred())
+			activeNumberOfFreeHugePages, err = strconv.Atoi(strings.Replace(buff.String(), "\r\n", "", 1))
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should alocate the amount of hugepages requested", func() {
+			Expect(activeNumberOfFreeHugePages).To(BeNumerically(">", 0))
+
+			pod := pods.RedefineWithHugePage(testDpdkNamespace, dpdkPod.Spec.NodeName, "200000000")
+			pod, err := client.Client.Pods(testDpdkNamespace).Create(pod)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() int {
+				buff, err := pods.ExecCommand(client.Client, *dpdkPod, []string{"cat",
+					fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-1048576kB/free_hugepages", numaNode)})
+				Expect(err).ToNot(HaveOccurred())
+				numberOfFreeHugePages, err := strconv.Atoi(strings.Replace(buff.String(), "\r\n", "", 1))
+				Expect(err).ToNot(HaveOccurred())
+
+				return numberOfFreeHugePages
+			}, 5*time.Minute, 5*time.Second).Should(Equal(activeNumberOfFreeHugePages - 3))
```

Comment:
Can you elaborate why we expect 3 less free hugepages? It's not clear by reading the code